### DISCUSSION TITLE
why use gap filling with carveme

### DISCUSSION THREAD

**Discussion by vrmarcelino** [franciscozorrilla/thesis_pipeline#39] [Q&A] [answered]
Hi! 

Thanks for sharing this pipeline!
I am trying to use it to understand microbial interactions, and I saw that carveme is run with the gap-filling flag. Could you explain why did you chose to use it by default? Considering that we are dealing with metagenomes, we can't be sure that the microbes can grow in isolation - perhaps they rely on essential nutrients provided by other members. On the other hand, MAGs are often incomplete and perhaps omitting gap-filling might overrepresent the interdependence between species. Have you tried both options?

Thanks!!

**Comment by franciscozorrilla** [comment, marked as answer]
Hi vrmarcelino,

Good question! Indeed, we cannot be sure that the microbes will grow in isolation, and the gapfilling is meant to ensure that the microbes can grow within the community. By default, the gapfilling is done with complete media (M8 in the config file/media_db.tsv, see figure 1d [here](https://www.nature.com/articles/s41564-018-0123-9/figures/1)), which assumes that the microbes *should* be able to grow in a community/environment where all nutrients are provided. In the metaGEM paper we gapfill on M8 and then simulate on M11 to observe exchanges associated with aromatic amino acids. Also note that the gapfill flag will use reaction confidence scores to prioritize the introduction of reactions/pathways with the most genetic evidence present in the contigs, i.e. not just adding random reactions until growth is achieved. Also, the log files of the carveme jobs will say how many metabolites/reactions are added from gapfilling, and from my experience it is usually none or in the low single digits.

Thanks for your interest in metaGEM and let me know if you have further questions!

**Comment by vrmarcelino** [comment]
Hi Francisco,

Thanks a lot for the explanation, that sounds very reasonable.

And thanks for the pipeline again!
